### PR TITLE
support alibi embedding

### PR DESCRIPTION
**Pull request by aurora327**

### CODE REVIEW

File: src/layers/alibi_embedding.cpp
[the remark this answers is not in the document]
Comment (aurora327) [reply src/layers/alibi_embedding.cpp:1]: Code has been formatted

**Comment by Duyi-Wang** [comment]
Please use Camel case to rename variables and functions.

File: src/layers/alibi_embedding.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    void alibiGetRelativePos(const int seqLen);
+
+    void alibiGetSlope(const int headNum);
+
+    // headIdx is [0,n]
+    void alibiGetBias(const int headIdx, const int seqLen, float *bias_matrx);
+
+private:
+    static bool initialized;
```
Comment (Duyi-Wang) [inline src/layers/alibi_embedding.h:23]: What's the meaning of this bool var? Nothing will be shared between objects.

Comment (aurora327) [reply src/layers/alibi_embedding.h:23]: This variable is used to determine whether the distance matrix has been initialized or not. And In this pr, 1. If alibiGetBias function is called directly without the initialization judgment, an error will occur. 2. The distance matrix can be reused w/ single instance.

Comment (Duyi-Wang) [reply src/layers/alibi_embedding.h:23]: All the functions are not static, so they can't be called without instantiating a class. And you will initialize it in the constructor. Will you offer a constructor method without initialization or a static method?

**Review by Duyi-Wang — APPROVED** [review]

**Pull request merged**